src/itest/java/uk/gov/companieshouse/company/metrics/steps/CompanyMetricsConsumerSteps.java

```diff
@@ -0,0 +1,94 @@
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.kafka.core.KafkaTemplate;
+import uk.gov.companieshouse.api.metrics.MetricsRecalculateApi;
+import uk.gov.companieshouse.stream.ResourceChangedData;
+
+public class CompanyMetricsConsumerSteps {
+
+    @Autowired
+    private TestSupport testSupport;
+
+    @Autowired
+    public KafkaTemplate<String, Object> kafkaTemplate;
+
+    @Autowired
+    protected TestRestTemplate restTemplate;
+
+    private WireMockServer wireMockServer;
+
+    /**
+     * The company number extracted from the current avro file
+     */
+    private String currentCompanyNumber;
+
+    @Given("Company Metrics Consumer component is running and Charges Data API is stubbed")
+    public void theCompanyMetricsIsRunning() {
+        ResponseEntity<String> response = restTemplate.getForEntity("/healthcheck", String.class);
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.valueOf(200));
+        assertThat(response.getBody()).isEqualTo("I am healthy");
+        wireMockServer = testSupport.setupWiremock();
+        assertThat(wireMockServer.isRunning()).isTrue();
+    }
+
+    @When("A valid avro message for {string} is generated and sent to the Kafka topic {string}")
+    public void generateAvroMessageSendToTheKafkaTopic(String companyNumber, String topic)
+        throws InterruptedException, IOException {
+        ResourceChangedData avroMessageData = testSupport.createResourceChangedMessage(companyNumber);
+        currentCompanyNumber = companyNumber;
+        stubCompanyMetricsApi(currentCompanyNumber, "an_avro_message_is_published_to_topic");
+        kafkaTemplate.send(topic, avroMessageData);
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        countDownLatch.await(5, TimeUnit.SECONDS);
+    }
+
+    @Then("The message is successfully consumed and company number is susccessfully extracted to call company-metrics-api POST endpoint with expected payload")
+    public void checkMessageComsumedAndCompanyMetricsApiCalledWithCorrectValues(){
+        List<ServeEvent> serverEvents = testSupport.getServeEvents();
+        assertThat(serverEvents.size()).isEqualTo(1);
+        assertThat(serverEvents.isEmpty()).isFalse(); // assert that the wiremock did something
+        assertThat(serverEvents.get(0).getRequest().getUrl()).isEqualTo(String.format("/company/%s/metrics/recalculate", currentCompanyNumber));
+        String body = new String(serverEvents.get(0).getRequest().getBody());
+        MetricsRecalculateApi payload = null;
+        ObjectMapper mapper = new ObjectMapper();
+        try {
+            payload = mapper.readValue(body, MetricsRecalculateApi.class);
+        } catch (JsonProcessingException e) {
+            e.printStackTrace();
+        }
+        assertThat(payload).isNotNull();
```

Comment:
Lets check if charges boolean flag is sent as true, and any other attribute we send in payload

Reply:
Additional asserts added